gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Project.java

```diff
@@ -26,17 +31,97 @@
  * immutable.  Methods that change or update the underlying Project information return a new Project
  * instance.
  */
-public class Project {
+public class Project extends ProjectInfo {
 
-  private final ResourceManager resourceManager;
-  private final ProjectInfo info;
+  private static final long serialVersionUID = 6767630161335155133L;
 
-  /**
-   * Constructs a Project object that contains the ProjectInfo given.
-   */
-  public Project(ResourceManager resourceManager, ProjectInfo projectInfo) {
+  private final ResourceManagerOptions options;
+  private transient ResourceManager resourceManager;
+
+  public static class Builder extends ProjectInfo.Builder {
+    private final ResourceManager resourceManager;
+    private ProjectInfo.BuilderImpl infoBuilder;
+
+    Builder(ResourceManager resourceManager) {
+      this.resourceManager = resourceManager;
+      this.infoBuilder = new ProjectInfo.BuilderImpl();
+    }
+
+    Builder(Project project) {
+      this.resourceManager = project.resourceManager;
+      this.infoBuilder = new ProjectInfo.BuilderImpl(project);
+    }
+
+    @Override
+    public Builder name(String name) {
+      infoBuilder.name(name);
+      return this;
+    }
+
+    @Override
+    public Builder projectId(String projectId) {
+      infoBuilder.projectId(projectId);
+      return this;
+    }
+
+    @Override
+    public Builder addLabel(String key, String value) {
+      infoBuilder.addLabel(key, value);
+      return this;
+    }
+
+    @Override
+    public Builder removeLabel(String key) {
+      infoBuilder.removeLabel(key);
+      return this;
+    }
+
+    @Override
+    public Builder clearLabels() {
+      infoBuilder.clearLabels();
+      return this;
+    }
+
+    @Override
+    public Builder labels(Map<String, String> labels) {
+      infoBuilder.labels(labels);
+      return this;
+    }
+
+    @Override
+    Builder projectNumber(Long projectNumber) {
+      infoBuilder.projectNumber(projectNumber);
+      return this;
+    }
+
+    @Override
+    Builder state(State state) {
+      infoBuilder.state(state);
+      return this;
+    }
+
+    @Override
+    Builder createTimeMillis(Long createTimeMillis) {
+      infoBuilder.createTimeMillis(createTimeMillis);
+      return this;
+    }
+
+    @Override
+    Builder parent(ResourceId parent) {
+      infoBuilder.parent(parent);
+      return this;
+    }
+
+    @Override
+    public Project build() {
+      return new Project(resourceManager, infoBuilder);
+    }
+  }
+
+  Project(ResourceManager resourceManager, ProjectInfo.BuilderImpl infoBuilder) {
```

Comment:
That's true, but having two parameters does make some other package-scope code slightly simpler.  For example, in [`Project.fromPb`](https://github.com/GoogleCloudPlatform/gcloud-java/pull/572/files#diff-f71225b79e2dfd12febf71cecffed53aR238), we avoid having to explicitly set all the fields of a `Project.Builder` using the `ProjectInfo`'s fields.  For that reason, I'm inclined to keep this function signature.